spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/JdbcAggregateTemplate.java

```diff
@@ -309,11 +313,43 @@ public void deleteAll(Class<?> domainType) {
 
 		entity = triggerBeforeDelete(entity, id, change);
 
-		executor.execute(change);
+		executor.executeDelete(change);
 
 		triggerAfterDelete(entity, id, change);
 	}
 
+	private <T> List<T> performSaveChange(Iterable<T> instances, Function<T, Function<T, AggregateChangeWithRoot<T>>> changeCreatorSelector) {
+
+		ArrayList<T> instancesList = new ArrayList<>();
```

Comment:
Typically, `instances` is already a collection, it would make sense to have a utility method to either cast or copy `instances` into a `List`